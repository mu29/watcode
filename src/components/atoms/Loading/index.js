import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { size, color } from 'services/style'

const Wrapper = styled.div`
  margin: ${({ inline }) => (inline ? 'auto 0.25rem' : 'auto')};
  padding: ${({ compact }) => (compact ? 0 : '1rem')};
`

const Progress = styled.div`
  width: ${size};
  height: ${size};
  margin: auto;
  animation: rotate 0.8s infinite linear;
  border: 0.125rem solid ${color};
  border-right-color: transparent;
  border-radius: 50%;

  @keyframes rotate {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`

const Loading = ({
  isLoading,
  inline,
  compact,
  children,
  ...props
}) => (
  isLoading ? (
    <Wrapper inline={ inline } compact={ compact }>
      <Progress { ...props } />
    </Wrapper>
  ) : children
)

Loading.propTypes = {
  isLoading: PropTypes.bool,
  children: PropTypes.node,
  color: PropTypes.string,
  size: PropTypes.number,
  compact: PropTypes.bool,
  inline: PropTypes.bool,
}

Loading.defaultProps = {
  isLoading: true,
  children: null,
  color: 'gray.100',
  size: 24,
  compact: false,
  inline: false,
}

export default Loading
