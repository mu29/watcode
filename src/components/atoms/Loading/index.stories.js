import React from 'react'
import { storiesOf } from '@storybook/react'
import Loading from '.'

storiesOf('Atoms/Loading', module)
  .add('기본', () => (
    <Loading color="black" visible />
  ))
