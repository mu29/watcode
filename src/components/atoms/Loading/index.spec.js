import React from 'react'
import Loading from '.'

const component = (props = {}) => withTheme(<Loading { ...props } />)

describe('<Loading />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = render(component())
    expect(wrapper).toMatchSnapshot()
  })
})
