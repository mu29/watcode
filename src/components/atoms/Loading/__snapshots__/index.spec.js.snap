// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> 정상적으로 렌더링된다 1`] = `
.c0 {
  margin: auto;
  padding: 1rem;
}

.c1 {
  width: 1.5rem;
  height: 1.5rem;
  margin: auto;
  -webkit-animation: rotate 0.8s infinite linear;
  animation: rotate 0.8s infinite linear;
  border: 0.1875rem solid;
  border-right-color: transparent;
  border-radius: 50%;
}

<div
  class="c0"
>
  <div
    class="c1"
    color="black"
    size="24"
  />
</div>
`;
